### PR TITLE
[Fix #12279] Fix false positives for `Lint/EmptyConditionalBody`

### LINKED ISSUE #12279: EmptyConditionalBody AllowedComments doesn't work for if/unless correctly
## Expected behavior

As documented here: https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/EmptyConditionalBody
``Checks for the presence of ‘if`, `elsif` and `unless` branches without a body.``
Expect `AllowComments` to work for both `if/unless` and `elsif` blocks

## Actual behavior

`AllowComments` works for `if/unless` blocks only when there is no `elsif/else` and code after the block ends at the same nesting level.
`AllowComments always works of `elsif` blocks

## Steps to reproduce the problem

Based on documentation all of these should be allowed when `AllowComments: true`
test code:
```
module Something

  def method1(foobar, barfoo)
    puts 'lets do something'
    if foobar # 1
      puts 'testing the conditional'
    end
    unless barfoo # 1
      # no error
    end
    puts 'foobar'
  end

  def method2(foobar, barfoo)
    puts 'lets do something'
    if foobar # 2
      puts 'testing the conditional'
    end
    unless barfoo # 2
      # issue found!
    end
    #puts 'foobar'
  end

  def method3(foobar, barfoo)
    puts 'lets do something'
    if foobar # 3
      # no error
    end
    unless barfoo # 3
      # issue found!
    end
  end

  def method4(foobar)
    puts 'lets do something'
    if foobar # 4
      # no error
    end
    puts 'foobar'
  end

  def method5(foobar)
    puts 'lets do something'
    if foobar # 5
      # issue found!
    end
  end

  def method6(foobar, barfoo)
    puts 'lets do something'
    if foobar # 6
      # no error
    elsif barfoo # 6
      # no error
    end
  end

  def method7(foobar)
    puts 'lets do something'
    if foobar # 7
      # no error
    else # 7
      puts 'hello'
    end
  end

  def method8(foobar, barfoo)
    puts 'lets do something'
    if foobar # 8
      puts 'foobar'
      if barfoo # 8
        # issue found!
      end
    end
    puts 'hello'
  end

  def method9(foobar, barfoo)
    puts 'lets do something'
    if foobar # 9
      puts 'foobar'
      if barfoo # 9
        # no error
      end
      puts 'hello'
    end
  end

end
```

test config:
```
require:
  - rubocop-performance
  - rubocop-rails

AllCops:
  TargetRubyVersion: 2.7

Lint/EmptyConditionalBody:
  AllowComments: true
  Enabled: true
```

test command and output:
```
rubocop --only Lint/EmptyConditionalBody test.rb
Inspecting 1 file
W

Offenses:

test.rb:19:5: W: [Correctable] Lint/EmptyConditionalBody: Avoid unless branches without a body.
    unless barfoo # 2 ...
    ^^^^^^^^^^^^^^^^^
test.rb:30:5: W: [Correctable] Lint/EmptyConditionalBody: Avoid unless branches without a body.
    unless barfoo # 3 ...
    ^^^^^^^^^^^^^^^^^
test.rb:45:5: W: [Correctable] Lint/EmptyConditionalBody: Avoid if branches without a body.
    if foobar # 5 ...
    ^^^^^^^^^^^^^
test.rb:72:7: W: [Correctable] Lint/EmptyConditionalBody: Avoid if branches without a body.
      if barfoo # 8 ...
      ^^^^^^^^^^^^^
```

There may be more weird edge cases, but this is as far down this rabbit-hole as I choose to go for now.

## RuboCop version

```
rubocop -V
1.57.1 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-darwin21]
  - rubocop-performance 1.19.1
  - rubocop-rails 2.21.2
```

### PR DESCRIPTION
**Pull request by koic**
Fixes #12279.

This PR fixes false positives for `Lint/EmptyConditionalBody` when missing 2nd `if` body with a comment.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/lint/empty_conditional_body_spec.rb [hunk 1]
```diff
@@ -347,10 +358,10 @@ class Foo
     it 'registers an offense for multi-line value omission in `unless`' do
       expect_offense(<<~RUBY)
         var =
+          # This is the value of `other:`, like so: `other: condition || other_condition`
           unless object.action value:, other:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid `unless` branches without a body.
-            condition || other_condition # This is the value of `other:`, like so:
-                                         # `other: condition || other_condition`
```
Comment (koic) [inline spec/rubocop/cop/lint/empty_conditional_body_spec.rb:353]: NOTE: This is an intentional change to treat it as an `if` that does not contain comments.

**Comment by bbatsov** [comment]
thanks!
